backend/src/test/java/com/phaete/backend/forage/service/ForageMapItemServiceTest.java

```diff
@@ -0,0 +1,335 @@
+						new ForageMapItem(
+								"1",
+								null,
+								new CustomMarker(
+										"1",
+										new double[] {0.0, 0.0},
+										"",
+										new int[] {0, 0},
+										new int[] {0, 0},
+										new int[] {0, 0},
+										""
+								),
+								ForageQuantity.ABUNDANT,
+								ForageQuality.EXCELLENT,
+								"never",
+								"notes"
+						)
+				)
+		);
+
+		ForageMapItemService forageMapItemService = new ForageMapItemService(
+				forageMapItemRepository,
+				new ConverterService(idService)
+		);
+
+		List<ForageMapItem> actualForageMapItems = forageMapItemService.findAllForageMapItems();
+		verify(forageMapItemRepository).findAll();
+		assertEquals(List.of(), actualForageMapItems);
+	}
+
+	@Test
+	void findAllForageMapItems_expectList_excludeItemsWithoutCustomMarkerInDB() {
+		when(forageMapItemRepository.findAll()).thenReturn(
+				List.of(
+						new ForageMapItem(
+								"1",
+								new ForageWikiItem(
+										"1",
+										"Apple Tree",
+										ForageCategory.FRUIT,
+										ForageSource.TREE,
+										"Apple Tree",
+										ForageSeason.FALL,
+										List.of("test")
+								),
+								null,
+								ForageQuantity.ABUNDANT,
+								ForageQuality.EXCELLENT,
+								"never",
+								"notes"
+						)
+				)
+		);
+
+		ForageMapItemService forageMapItemService = new ForageMapItemService(
+				forageMapItemRepository,
+				new ConverterService(idService)
+		);
+
+		List<ForageMapItem> actualForageMapItems = forageMapItemService.findAllForageMapItems();
+		verify(forageMapItemRepository).findAll();
+		assertEquals(List.of(), actualForageMapItems);
+	}
+
+	@Test
+	void findForageMapItemById_expectDTO_onSuccess()
+			throws ForageMapItemNotFoundException, MarkerNotFoundException, ForageWikiItemNotFoundException {
+		ForageMapItemDTO expectedForageMapItemDTO = new ForageMapItemDTO(
+				new ForageWikiItem(
+						"1",
+						"Apple Tree",
+						ForageCategory.FRUIT,
+						ForageSource.TREE,
+						"Apple Tree",
+						ForageSeason.FALL,
+						List.of("test")
+				),
+				new CustomMarker(
+						"1",
+						new double[] {0.0, 0.0},
+						"",
+						new int[] {0, 0},
+						new int[] {0, 0},
+						new int[] {0, 0},
+						""
+				),
+				ForageQuantity.ABUNDANT,
+				ForageQuality.EXCELLENT,
+				"never",
+				"notes"
+		);
+
+		when(forageMapItemRepository.findById("1")).thenReturn(
+				Optional.of(
+						new ForageMapItem(
+								"1",
+								new ForageWikiItem(
+										"1",
+										"Apple Tree",
+										ForageCategory.FRUIT,
+										ForageSource.TREE,
+										"Apple Tree",
+										ForageSeason.FALL,
+										List.of("test")
+								),
+								new CustomMarker(
+										"1",
+										new double[] {0.0, 0.0},
+										"",
+										new int[] {0, 0},
+										new int[] {0, 0},
+										new int[] {0, 0},
+										""
+								),
+								ForageQuantity.ABUNDANT,
+								ForageQuality.EXCELLENT,
+								"never",
+								"notes"
+						)
+				)
```

Comment:
better create variables forageWikiItem, customMarker and forageMapItem before line 222. then the when-Statement will be clearer

Reply:
I don't quite understand what the problem is here. Everything is clearly separated with indentations making it clear that the Optional-Object I am building here is the supposed return value for the when-call. 
Why do I need to store everything in a variable before I am allowed to use it?

Reply:
The problem is that one has to inspect intensively if there are differences between this object and the one you expect to be returned. And if so, are they intented or typos? ("Do not repeat yourself!")

When you put a variable "item" in and expect the same variable as result, it would be much easier to see the relevant part of the test.